### ISSUE TITLE
GEA2 support

### ISSUE THREAD

**Issue by ablyler** [geappliances/home-assistant-adapter#5]
I’m curious if this adapter or code supports GEA2. I noticed it’s supported in the circuit diagram, but I couldn’t find any connections to it in the code. Any insights would be great!

**Comment by ryanplusplus** [comment]
Hey @ablyler, right now the hardware supports GEA2 but the code does not.

Support for GEA2 is being added to the underlying libraries, but we don't have any near term plans to add direct support in the adapter firmware because GEA2 doesn't allow us to discover which ERDs are supported like we can with GEA3 and we won't be able to maintain ERD lists for every GEA2 model. However, once we have library support for GEA2, there won't be anything stopping you from making a custom firmware that is tailored to a particular GEA2 appliance (ie: has a hard-coded list of supported ERDs).

I'm also working on some libraries to expose GEA2 and GEA3 capabilities with a more conventional Arduino API. These would let you more easily interact with your GEA2 or GEA3 appliance in a custom project, with or without HA.

Once we have all of the library support in place, it may make sense for us to maintain a separate firmware for GEA2 that makes it easy to specify which ERDs it should expose to HA, but I can't make any promises. For now we're focused on making support for GEA3 appliances as good as it can be.

**Comment by ablyler** [comment]
Thanks for the quick response @ryanplusplus! That is great news all around! I am really looking forward to this project. Let me know if I can be of any assistance, especially with the GEA2 track.

One question, regarding the hardware spec. Is the M74VHC1G07 component in the GEA2 bus missing a "C" and ment to be MC74VHC1G07 from OnSemi?

Thanks again for all the work on this. It is really impressive!

**Comment by ryanplusplus** [comment]
> One question, regarding the hardware spec. Is the M74VHC1G07 component in the GEA2 bus missing a "C" and meant to be MC74VHC1G07 from OnSemi?

@gillspice can you answer this?

**Comment by gillspice** [comment]
Yes MC74VHC1G07 or equivalent (74LVC1G07, SN74LVC1G07) is valid.

**Comment by paulgoodjohn** [comment]
> [@ryanplusplus](https://github.com/ryanplusplus) I'm looking for some general guidance on GEA2. I've taken the list of supported ERDs (from [here](https://github.com/geappliances/public-appliance-api-documentation/blob/main/appliance_api_erd_definitions.json)) and I have some code that reads each one and records which ones provide a response. This "initialization" process takes about 2+ hours but only needs to happen once. The known addresses are then stored in NVM.
> 
> Is there a way to subscribe to those known-good addresses or do I have to poll them?
> 
> And do you have suggestions for a better way to go about this?

The biggest difference in my version from your code (which by the way was the spur to my one) is that all appliances don't have all erds ...so for example a fridge will have erds from 0x0xxx range (common), 0x1xxx (refrigeration) and possibly 0xdxxx (energy) but it won't have any from the water heater space 0x4xxx. That cuts the time down quite a lot.
The other difference is that mine uses the 0x0008 response to work out the gea address ...only one board should respond to that....
Hope this helps

**Comment by mekaneck** [comment]
> The biggest difference in my version from your code (which by the way was the spur to my one) is that all appliances don't have all erds ...so for example a fridge will have erds from 0x0xxx range (common), 0x1xxx (refrigeration) and possibly 0xdxxx (energy) but it won't have any from the water heater space 0x4xxx. That cuts the time down quite a lot. The other difference is that mine uses the 0x0008 response to work out the gea address ...only one board should respond to that.... Hope this helps

I see that now, it's awesome that you are able to split up the ErdLists file based on appliance. It's exciting to see what someone can do when they know both the code and the system its being applied to. I know neither 😅. You seem to be making progress at lightning speed, I'll keep an eye on this and hopefully I can mothball my work in the not-too-distant future. Thank you for your efforts!

**Comment by jorgeluiso** [comment]
How can I install this to test?

**Comment by paulgoodjohn** [comment]
> How can I install this to test?

Well, if you go to my fork of this repo here:
https://github.com/paulgoodjohn/home-assistant-adapter
You should clone the repo, and set up the same way as this one is set up. My recommendation is that you use VSCode and platformIO to build and install it (it's what I use) so essentially it's a case of plugging the adapter into the PC with a USB cable, then clicking on the -> in the bottom of the window....
